src/transitions/modal-transition.js

```diff
@@ -0,0 +1,249 @@
+			callback: function () {
+				beginCompleted = true;
+				if (loadCompleted) {
+					completeAnim(dataIn);
+				}
+			}
+		});
+	};
+	
+	var defaultTransition = function (data, callback) {
+		dataIn = data;
+		loadCompleted = true;
+		App.modules.notify('site.removeScroll');
+		if (beginCompleted) {
+			completeAnim(data, callback);
+		}
+	};
+	
+	/* LEAVING FUNCTIONS */
+	var completeLeaveAnim = function (data, callback) {
+		var leavingPage = data.currentPage;
+		var enteringPage = data.nextPage;
+		
+		var domEnteringPage = $(enteringPage.key());
+		var domLeavingPage = $(leavingPage.key());
+		
+		var modal = domLeavingPage.find(MODAL_SELECTOR);
+		
+		var isModalReturn = domEnteringPage.hasClass('is-bg-modal-page');
+		
+		// Leave the current page
+		leavingPage.leave(data.leaveCurrent, {
+			canRemove: true
+		});
+		
+		// Make the page hidden for assistive technology
+		domLeavingPage.attr('aria-hidden', 'true').attr('role', 'presentation');
+		domEnteringPage.removeAttr('aria-hidden').removeAttr('role');
+		
+		//Check if the enteringPage is the background page
+		if (!isModalReturn) {
+			//Remove the other pages
+			var oldBgPage = sitePages.find('.is-bg-modal-page');
+			if (oldBgPage.length) {
+				body.removeClass(oldBgPage.attr('id'));
+				oldBgPage.remove();
+			}
+			
+			//Add the new class to the body
+			body.addClass(enteringPage.key().substring(1));
+		}
+		
+		// Notify entering page
+		App.modules.notify('page.entering', {page: enteringPage, route: data.route});
+		
+		domEnteringPage.ready(function () {
+			domLeavingPage.hide();
+			domEnteringPage.css({
+				opacity: 1,
+				display: 'block',
+				position: 'relative'
+			}).height();
+			enteringPage.enter(data.enterNext);
+			
+			App.modules.notify('modal.hideBg', {
+				callback: function () {
+					App.modules.notify('transition.end', {page: enteringPage, route: data.route});
+					App.modules.notify('site.addScroll');
+					App.modules.notify('transition.finished', {
+						page: enteringPage,
+						route: data.route
+					});
+				}
+			});
+			
+			App.callback(callback);
+		});
+	};
+	
+	var defaultLeaveBeginTransition = function (data, callback) {
+		var leavingPage = data.currentPage;
+		var domLeavingPage = $(leavingPage.key());
+		var modal = domLeavingPage.find(MODAL_SELECTOR);
+		
+		beginCompleted = false;
+		loadCompleted = false;
+		dataIn = null;
+		
+		App.modules.notify('modal.hide', {
+			item: modal,
+			callback: function () {
+				body.removeClass(leavingPage.key().substring(1));
+				App.modules.notify('page.leaving', {
+					page: leavingPage,
+					canRemove: true
+				});
+				
+				beginCompleted = true;
+				
+				if (loadCompleted) {
+					completeLeaveAnim(dataIn);
+				}
+			}
+		});
+	};
+	
+	var defaultLeaveTransition = function (data, callback) {
+		loadCompleted = true;
+		dataIn = data;
+		if (beginCompleted) {
+			completeLeaveAnim(data, callback);
+		}
+	};
+	
+	/*App.transitions.exports({
```

Comment:
We should maybe clean this a bit one example is enough 😁

Reply:
@dominic-blain  ?

Reply:
I say we let one example, right now their's 2 of them.

Reply:
Yes, my point exactly